### PR TITLE
Use cache in make task info

### PR DESCRIPTION
**Pull request by ujiuji1259**
I've added the `cache` argument to `make_task_info_tree` to use cache in making TaskInfo tree.

This reduces time to make TaskInfo tree especially when the task tree is large.

### CODE REVIEW

File: gokart/tree/task_info_formatter.py [hunk 1]
```diff
@@ -56,14 +56,21 @@ def _make_requires_info(requires):
...
-def make_task_info_tree(task: TaskOnKart, ignore_task_names: Optional[List[str]] = None) -> TaskInfo:
+def make_task_info_tree(task: TaskOnKart, ignore_task_names: Optional[List[str]] = None, cache: Optional[Dict[str, TaskInfo]] = None) -> TaskInfo:
     with warnings.catch_warnings():
         warnings.filterwarnings(action='ignore', message='Task .* without outputs has no custom complete() method')
         is_task_complete = task.complete()
 
     name = task.__class__.__name__
     unique_id = task.make_unique_id()
     output_paths = [t.path() for t in luigi.task.flatten(task.output())]
+
```
Comment (hirosassa) [inline gokart/tree/task_info_formatter.py:67]: [nits] redundant line

**Review by hirosassa — CHANGES_REQUESTED** [review]
Thanks for your contribution! I added a small comment, but almost LGTM

**Review by hirosassa — APPROVED** [review]
LGTM!

**Comment by mski-iksm** [comment]
@ujiuji1259 LGTM!

**Review by mski-iksm — APPROVED** [review]

**Comment by Hi-king** [comment]
@ujiuji1259 Thx for your contribution, @mski-iksm @hirosassa thx for reviews :)

**Pull request merged**